crates/rpc/rpc-eth-types/src/cache/mod.rs

```diff
@@ -186,6 +182,21 @@ impl<B: Block, R: Send + Sync> EthStateCache<B, R> {
         let _ = self.to_service.send(CacheAction::GetHeader { block_hash, response_tx });
         rx.await.map_err(|_| ProviderError::CacheServiceUnavailable)?
     }
+
+    /// Returns the most recent canonical block from the cache, if available.
+    /// Used to efficiently handle latest block requests and avoid race conditions during chain
+    /// reorgs.
+    /// Returns `None` if no canonical chain is tracked or during reorgs.
+    pub async fn latest_block_with_senders(
+        &self,
+    ) -> ProviderResult<Option<Arc<SealedBlockWithSenders<B>>>> {
+        if let Some(chain_change) = &self.latest_chain_change {
+            if let Some(latest_block) = chain_change.blocks.last() {
+                return self.get_sealed_block_with_senders(latest_block.hash()).await;
+            }
+        }
```

Comment:
this should always send an action to the service, but for fetching the latest, we'd need a new action variant